tools/pyi/gen_pyi.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import print_function
+import os
+import sys
+import inspect
+
+needed_modules = set()
+
+
+def type_to_python(typename):
```

Comment:
I know you plan to clean this up eventually, but it would be nice to have docstrings on all of these functions.